### DISCUSSION TITLE
MRTK3: Quest/Quest Pro Passthrough Support

### DISCUSSION THREAD

**Discussion by trzy** [microsoft/MixedRealityToolkit-Unity#11230] [Q&A]
Hi,

I'm evaluating MRTK3 and like what I see so far -- great job everyone! -- but am wondering if it is possible to enable passthrough mode for Quest apps?

I'm rather unfamiliar with how the OpenXR integration works. Normally, when importing the Oculus integration, it asks whether to use OpenXR but it seems to make no difference if this is selected. In the XR plugin provider, the Oculus plugin is the only option when building for Android.

With MRTK3, this is not the case, and even after importing the Oculus integration, that is not an option. Does that mean these Oculus-specific features are not possible to use?

Thanks,

Bart

**Comment by trzy** [comment]
I notice that this question has partly been addressed elsewhere. I decided to do a test and created a new project with Oculus Integration and then imported the minimal set of MRTK3 packages (input and UI elements). The Android build is configured to use the Oculus XR plugin (note that OpenXR wasn't even an option now, I'm not sure how to show to enable this as a selection). 

I can take an MRTK XR Rig and add the OVRManager and OVRPassthroughLayer scripts to it and it seems to work fine, including with passthrough enabled. The only issue is that hand tracking doesn't properly work. I enable both controllers and hands on OVRManager and when I put my controller down, I can see that the pointer ray disappears and Oculus hand tracking has been enabled (I can see the system menus on my hands) but MRTK is clearly not seeing the hands and they are unusable.

Is this because hand tracking requires the OpenXR plugin? Is there any way to make it work with the Oculus plugin?

**Reply by Zee2** [reply]
Hey @trzy we're currently working hard on getting passthrough shipped in a nice way. There are currently some issues with how the OIP plugin fights with other Unity APIs. It's not exactly a well-design system... there's lots of conflicts with other hand tracking APIs and it's generally making life quite difficult for us 😅 Stay tuned, we should have news on this soon.
